commit 913f679d5a4a3c4d0c6916cbdd065569945dc2a6
Author: Joyee Cheung <joyee@igalia.com>
Date:   Mon Apr 6 18:15:41 2026 +0200

    Fix non-conforming braced-init-list in value_or()
    
    value_or() is a function template that deduces its argument type.
    Passing a braced-init-list {} causes deduction to fail in GCC
    because {} has no type:
    
      wasm-shuffle-reducer.cc:576: error: no matching member function
      for call to 'value_or'
      note: candidate template ignored: couldn't infer template argument
    
    Clang only accepted it by chance. Replace {} with uint8_t{0} to make
    the type explicit.
    
    Refs: https://eel.is/c++draft/temp.deduct.call
    Refs: https://github.com/nodejs/node/pull/62572
    Change-Id: I589617c78fbc1a65a3475957f846ffb87364ff28
    Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/7726131
    Reviewed-by: Matthias Liedtke <mliedtke@chromium.org>
    Commit-Queue: Joyee Cheung <joyee@igalia.com>
    Cr-Commit-Position: refs/heads/main@{#106357}


--- a/v8/src/compiler/turboshaft/wasm-shuffle-reducer.cc
+++ b/v8/src/compiler/turboshaft/wasm-shuffle-reducer.cc
@@ -573,7 +573,7 @@ void WasmShuffleAnalyzer::TryReduceFromM
     uint8_t index = shuffle.shuffle[i];
     if (index >= lower_limit && index <= upper_limit) {
       max = std::max(static_cast<uint8_t>(index % kSimd128Size),
-                     max.value_or({}));
+                     max.value_or(uint8_t{0}));
     }
   }
   if (max) {
